### PR TITLE
Adding regex check for package directory names

### PR DESCRIPTION
**Pull request by takirala**
Currently the build.sh validates a package name that is like this : 

```bash

└── repo/package
    ├── A
    │   ├── hello-world-valid
    ├── AAA
    │   ├── Hello World Invalid
    │   ├── hello_world_invalid
    │   ├── 9numberhello-invalid
    │   ├── hello-world-invaliD
    └── ...
```

- This commit invalidates all the packages that are invalid. 
- directories under the package/ folder should only be [A-Z]
- Regex for folder name is ("^[a-z][a-z0-9-]+[a-z0-9]$")

### CODE REVIEW

**Comment by takirala** [comment]
@jsancio @cruhland Should I change the regex from 

`("^[a-z][a-z0-9-]+[a-z0-9]$")` to
`("^[a-z][a-z0-9-]*[a-z0-9]$")` 

The first one forces the user to have a name that is of at least length 3. I feel it is a sensible threshold. Comments?

File: scripts/validate-packages.py [hunk 1]
```diff
@@ -3,13 +3,16 @@
...
+import re
 import sys
 from distutils.version import LooseVersion
 
 SCRIPTS_DIR = os.path.dirname(os.path.realpath(__file__))
 UNIVERSE_DIR = os.path.join(SCRIPTS_DIR, "..")
 PKG_DIR = os.path.join(UNIVERSE_DIR, "repo/packages")
 SCHEMA_DIR = os.path.join(UNIVERSE_DIR, "repo/meta/schema")
+LETTER_PATTERN = re.compile("^[A-Z]$")
+PACKAGE_FOLDER_PATTERN = re.compile("^[a-z][a-z0-9-]+[a-z0-9]$")
```
Comment (cruhland) [inline scripts/validate-packages.py]: @takirala regarding your question: yes, let's change the `+` to a `*`. There might be a package name that's only two characters long; consider the S3 object store or the H2 in-memory database.

File: scripts/validate-packages.py [hunk 2]
```diff
@@ -30,6 +33,11 @@ def _get_json_schema(file_name):
 def main():
     # traverse prefix dirs ("A", "B", etc)
     for letter in os.listdir(PKG_DIR):
+        if not LETTER_PATTERN.match(letter):
+            sys.exit(
+                "\tERROR\n\n"
+                "Invalid name for directory : {}".format(letter)
```
Comment (cruhland) [inline scripts/validate-packages.py]: Reword the error message to include the regex that the directory must match; in general, error messages should include information to help fix the problem, where possible.

File: scripts/validate-packages.py [hunk 3]
```diff
@@ -125,15 +135,25 @@ def _validate_revision(given_package, revision, path):
...
+    if (old_package and resource_json and 'cli' in resource_json and
             command_json is None):
         sys.exit('\tERROR\n\nA package with CLI specified in resource.json is '
                  'only supported when minDcosReleaseVersion is greater than '
                  '1.8.')
 
 
+def _validate_package_with_directory(given_package, actual_package_name):
+    if(not PACKAGE_FOLDER_PATTERN.match(given_package)
+       or given_package != actual_package_name):
```
Comment (cruhland) [inline scripts/validate-packages.py]: Let's have a separate check and error message for each condition, again providing helpful information for debugging.

**Comment by takirala** [comment]
@cruhland 

I have updated the error message to look like this:

Package Letter Directory Error:
```bash
Invalid name for directory : Test
Name should match the pattern ^[A-Z]$
```

Invalid name for folder due to regex:
```bash
Invalid name for package directory : tunnel-Acli. Name should matchthe pattern : ^[a-z][a-z0-9-]*[a-z0-9]$
```

Invalid name for folder due to mismatch:
```bash
The name parameter in package.json should match with the name of the package directory.
Directory : tunnel-cli2, Parsed Name : tunnel-cli
```

**Comment by cruhland** [comment]
:ship: it @takirala!

**Pull request merged**